From 362c1d173c155a259a32f7cd9c70ea375b503cef Mon Sep 17 00:00:00 2001
From: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Date: Thu, 9 Jan 2025 19:10:28 +0100
Subject: [PATCH 772/954] feat(stm32mp2): increment HDPL before booting BL31

Increment HDPL at the end of function bl2_el3_plat_prepare_exit to
execute BL31 in HDPL2.

Upstream-Status: Inappropriate [ST downstream https://github.com/STMicroelectronics/arm-trusted-firmware.git e7b068e80102f7880be9144fb474951d7930aa9d feat(stm32mp2): increment HDPL before booting BL31]
Change-Id: I35f111feeeaa5669c95601105077aa0aef0f4b93
Signed-off-by: Thomas Bourgoin <thomas.bourgoin@foss.st.com>
Reviewed-on: https://gerrit.st.com/c/mpu/oe/st/tf-a/+/427990
ACI: CIBUILD <MDG-smet-aci-builds@list.st.com>
Reviewed-by: Maxime MERE <maxime.mere@foss.st.com>
Domain-Review: Yann GAUTIER <yann.gautier@foss.st.com>
ACI: CITOOLS <MDG-smet-aci-reviews@list.st.com>
---
 plat/st/stm32mp2/bl2_plat_setup.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/plat/st/stm32mp2/bl2_plat_setup.c b/plat/st/stm32mp2/bl2_plat_setup.c
index 9fdc28e5b0..b64f563990 100644
--- a/plat/st/stm32mp2/bl2_plat_setup.c
+++ b/plat/st/stm32mp2/bl2_plat_setup.c
@@ -776,4 +776,7 @@ void bl2_el3_plat_prepare_exit(void)
 
 	/* Unmask potential tamper before exit */
 	stm32mp_syscfg_mask_potential_tamper_disable();
+
+	/* Increment hide protection level */
+	bsec_increment_hdpl();
 }
-- 
2.53.0

